test_session_suspend_resume times out waiting for Active state after resume

## Summary

`test_session_suspend_resume[chromium]` consistently times out after ~90 seconds and is skipped with:

```
Session did not return to Active state after resume
```

Suspend and resume works correctly when tested manually via the Workbench UI.

## Root Cause Analysis

The test in `src/vip_tests/workbench/test_sessions.py` (lines 159-171):

1. Suspends the session
2. Waits for "Suspended" state (succeeds)
3. Resumes the session
4. Navigates back to the homepage via `page.go_back()` or `page.goto()`
5. Polls `div[aria-label='Active']` via Playwright's `expect().to_be_visible(timeout=90000)`

The problem is in step 4-5: the test navigates back to the homepage but **does not force a page reload**. The Workbench homepage updates session state via WebSocket/background polling, but after `page.go_back()`, the cached DOM may still show the stale "Suspended" status. Playwright's `to_be_visible()` polls the existing DOM — if the frontend's own refresh hasn't fired, the selector never matches within the 90-second timeout.

## Why It Works in the UI

When a user manually tests suspend/resume, they interact with the page (clicking, scrolling) which triggers re-renders and API fetches that update the session state display. The automated test has no such interactions.

## Suggested Fix

Add a `page.reload()` before polling for the Active state, or implement explicit API polling against the Workbench sessions endpoint to wait for the state transition before checking the DOM.

## Reproduction

```bash
vip verify --workbench-url https://localhost --headless-auth --insecure
# test_session_suspend_resume[chromium]
#   ... still running (30s)
#   ... still running (60s)
#   ... still running (90s)
# SKIPPED (Session did not return to Active state...)
```

Console History: 

```
root@instance-20260430-1777562952-pwb-01:~# useradd --create-home --home-dir /home/testuser -s /bin/bash testuser;
echo -e 'test123\ntest123' | suroot@instance-20260430-1777562952-pwb-01:~# echo -e 'test123\ntest123' | sudo passwd testuser;
New password: Retype new password: passwd: password updated successfully
root@instance-20260430-1777562952-pwb-01:~#
root@instance-20260430-1777562952-pwb-01:~#
root@instance-20260430-1777562952-pwb-01:~#
root@instance-20260430-1777562952-pwb-01:~# uv venv
source .venv/bin/activate
uv pip install posit-vip
uv run playwright install --with-deps chromium
vip verify --connect-url https://connect.example.com --interactive-auth
Using CPython 3.12.3 interpreter at: /usr/bin/python3
Creating virtual environment at: .venv
Activate with: source .venv/bin/activate
Resolved 66 packages in 400ms
Prepared 66 packages in 892ms
Installed 66 packages in 32ms
 + annotated-types==0.7.0
 + anyio==4.13.0
 + asgiref==3.11.1
 + brand-yml==0.1.1
 + certifi==2026.4.22
 + charset-normalizer==3.4.7
 + click==8.3.3
 + eval-type-backport==0.3.1
 + execnet==2.1.2
 + gherkin-official==29.0.0
 + greenlet==3.5.0
 + h11==0.16.0
 + htmltools==0.6.0
 + httpcore==1.0.9
 + httpx==0.28.1
 + idna==3.13
 + importlib-metadata==8.7.1
 + iniconfig==2.3.0
 + libsass==0.23.0
 + linkify-it-py==2.1.0
 + mako==1.3.12
 + markdown-it-py==4.0.0
 + markupsafe==3.0.3
 + mdit-py-plugins==0.5.0
 + mdurl==0.1.2
 + narwhals==2.20.0
 + opentelemetry-api==1.41.1
 + orjson==3.11.8
 + packaging==26.2
 + parse==1.21.1
 + parse-type==0.6.6
 + pip==26.1
 + platformdirs==4.9.6
 + playwright==1.59.0
 + pluggy==1.6.0
 + posit-vip==0.28.0
 + prompt-toolkit==3.0.52
 + pydantic==2.13.3
 + pydantic-core==2.46.3
 + pyee==13.0.1
 + pygments==2.20.0
 + pytest==9.0.3
 + pytest-base-url==2.1.0
 + pytest-bdd==8.1.0
 + pytest-playwright==0.7.2
 + pytest-xdist==3.8.0
 + python-multipart==0.0.27
 + python-slugify==8.0.4
 + questionary==2.1.1
 + requests==2.33.1
 + ruamel-yaml==0.19.1
 + setuptools==82.0.1
 + shiny==1.6.0
 + shinychat==0.3.0
 + six==1.17.0
 + starlette==1.0.0
 + text-unidecode==1.3
 + typing-extensions==4.15.0
 + typing-inspection==0.4.2
 + uc-micro-py==2.0.0
 + urllib3==2.6.3
 + uvicorn==0.46.0
 + watchfiles==1.1.1
 + wcwidth==0.6.0
 + websockets==16.0
 + zipp==3.23.1
Installing dependencies...
Hit:1 http://us-east-2.ec2.archive.ubuntu.com/ubuntu noble InRelease
Get:2 http://us-east-2.ec2.archive.ubuntu.com/ubuntu noble-updates InRelease [126 kB]
Get:3 http://us-east-2.ec2.archive.ubuntu.com/ubuntu noble-backports InRelease [126 kB]
Get:4 http://us-east-2.ec2.archive.ubuntu.com/ubuntu noble-updates/main amd64 Packages [1946 kB]
Get:5 http://us-east-2.ec2.archive.ubuntu.com/ubuntu noble-updates/main Translation-en [348 kB]
Get:6 http://us-east-2.ec2.archive.ubuntu.com/ubuntu noble-updates/universe amd64 Packages [1685 kB]
Get:7 http://us-east-2.ec2.archive.ubuntu.com/ubuntu noble-updates/restricted amd64 Packages [3095 kB]
Get:8 http://us-east-2.ec2.archive.ubuntu.com/ubuntu noble-updates/restricted Translation-en [715 kB]
0% [Waiting for headers]                                                                                                                                                                                                                                                                                                                 1252 kB/s 0s                                                                                                                                                                 1252 kB0% [Waiting for headers]
0% [Waiting for headers]
0% [Waiting for headers]
Ign:9 http://security.ubuntu.com/ubuntu noble-security InRelease
Get:9 http://security.ubuntu.com/ubuntu noble-security InRelease [126 kB]
Get:10 http://security.ubuntu.com/ubuntu noble-security/main amd64 c-n-f Metadata [11.0 kB]
Fetched 8179 kB in 1min 49s (75.2 kB/s)
Reading package lists... Done
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
libasound2t64 is already the newest version (1.2.11-1ubuntu0.2).
libasound2t64 set to manually installed.
libatk-bridge2.0-0t64 is already the newest version (2.52.0-1build1).
libatk-bridge2.0-0t64 set to manually installed.
libatk1.0-0t64 is already the newest version (2.52.0-1build1).
libatk1.0-0t64 set to manually installed.
libatspi2.0-0t64 is already the newest version (2.52.0-1build1).
libatspi2.0-0t64 set to manually installed.
libcairo2 is already the newest version (1.18.0-3build1).
libcups2t64 is already the newest version (2.4.7-1.2ubuntu7.9).
libcups2t64 set to manually installed.
libdbus-1-3 is already the newest version (1.14.10-4ubuntu4.1).
libdbus-1-3 set to manually installed.
libdrm2 is already the newest version (2.4.125-1ubuntu0.1~24.04.1).
libdrm2 set to manually installed.
libgbm1 is already the newest version (25.2.8-0ubuntu0.24.04.1).
libgbm1 set to manually installed.
libglib2.0-0t64 is already the newest version (2.80.0-6ubuntu3.8).
libglib2.0-0t64 set to manually installed.
libnspr4 is already the newest version (2:4.35-1.1build1).
libnspr4 set to manually installed.
libnss3 is already the newest version (2:3.98-1ubuntu0.1).
libnss3 set to manually installed.
libpango-1.0-0 is already the newest version (1.52.1+ds-1build1).
libpango-1.0-0 set to manually installed.
libx11-6 is already the newest version (2:1.8.7-1build1).
libx11-6 set to manually installed.
libxcb1 is already the newest version (1.15-1ubuntu2).
libxcb1 set to manually installed.
libxcomposite1 is already the newest version (1:0.4.5-1build3).
libxcomposite1 set to manually installed.
libxdamage1 is already the newest version (1:1.1.6-1build1).
libxdamage1 set to manually installed.
libxext6 is already the newest version (2:1.3.4-1build2).
libxext6 set to manually installed.
libxfixes3 is already the newest version (1:6.0.0-2build1).
libxfixes3 set to manually installed.
libxkbcommon0 is already the newest version (1.6.0-1build1).
libxkbcommon0 set to manually installed.
libxrandr2 is already the newest version (2:1.5.2-2build1).
libxrandr2 set to manually installed.
xvfb is already the newest version (2:21.1.12-1ubuntu1.5).
libfontconfig1 is already the newest version (2.15.0-1.1ubuntu2).
libfreetype6 is already the newest version (2.13.2+dfsg-1ubuntu0.1).
Recommended packages:
  fonts-ipafont-mincho fonts-liberation-sans-narrow fonts-tlwg-loma
The following NEW packages will be installed:
  fonts-freefont-ttf fonts-ipafont-gothic fonts-liberation fonts-noto-color-emoji fonts-tlwg-loma-otf fonts-unifont fonts-wqy-zenhei xfonts-cyrillic xfonts-scalable
0 upgraded, 9 newly installed, 0 to remove and 16 not upgraded.
Need to get 31.8 MB of archives.
After this operation, 93.6 MB of additional disk space will be used.
Get:1 http://us-east-2.ec2.archive.ubuntu.com/ubuntu noble/universe amd64 fonts-ipafont-gothic all 00303-21ubuntu1 [3513 kB]
Get:2 http://us-east-2.ec2.archive.ubuntu.com/ubuntu noble/main amd64 fonts-freefont-ttf all 20211204+svn4273-2 [5641 kB]
Get:3 http://us-east-2.ec2.archive.ubuntu.com/ubuntu noble/main amd64 fonts-liberation all 1:2.1.5-3 [1603 kB]
Get:4 http://us-east-2.ec2.archive.ubuntu.com/ubuntu noble-updates/main amd64 fonts-noto-color-emoji all 2.047-0ubuntu0.24.04.1 [9764 kB]
Get:5 http://us-east-2.ec2.archive.ubuntu.com/ubuntu noble/universe amd64 fonts-tlwg-loma-otf all 1:0.7.3-1 [107 kB]
Get:6 http://us-east-2.ec2.archive.ubuntu.com/ubuntu noble/universe amd64 fonts-unifont all 1:15.1.01-1build1 [2993 kB]
Get:7 http://us-east-2.ec2.archive.ubuntu.com/ubuntu noble/universe amd64 fonts-wqy-zenhei all 0.9.45-8 [7472 kB]
Get:8 http://us-east-2.ec2.archive.ubuntu.com/ubuntu noble/universe amd64 xfonts-cyrillic all 1:1.0.5+nmu1 [384 kB]
Get:9 http://us-east-2.ec2.archive.ubuntu.com/ubuntu noble/main amd64 xfonts-scalable all 1:1.0.3-1.3 [304 kB]
Fetched 31.8 MB in 1s (51.7 MB/s)
Selecting previously unselected package fonts-ipafont-gothic.
(Reading database ... 207890 files and directories currently installed.)
Preparing to unpack .../0-fonts-ipafont-gothic_00303-21ubuntu1_all.deb ...
Unpacking fonts-ipafont-gothic (00303-21ubuntu1) ...
Selecting previously unselected package fonts-freefont-ttf.
Preparing to unpack .../1-fonts-freefont-ttf_20211204+svn4273-2_all.deb ...
Unpacking fonts-freefont-ttf (20211204+svn4273-2) ...
Selecting previously unselected package fonts-liberation.
Preparing to unpack .../2-fonts-liberation_1%3a2.1.5-3_all.deb ...
Unpacking fonts-liberation (1:2.1.5-3) ...
Selecting previously unselected package fonts-noto-color-emoji.
Preparing to unpack .../3-fonts-noto-color-emoji_2.047-0ubuntu0.24.04.1_all.deb ...
Unpacking fonts-noto-color-emoji (2.047-0ubuntu0.24.04.1) ...
Selecting previously unselected package fonts-tlwg-loma-otf.
Preparing to unpack .../4-fonts-tlwg-loma-otf_1%3a0.7.3-1_all.deb ...
Unpacking fonts-tlwg-loma-otf (1:0.7.3-1) ...
Selecting previously unselected package fonts-unifont.
Preparing to unpack .../5-fonts-unifont_1%3a15.1.01-1build1_all.deb ...
Unpacking fonts-unifont (1:15.1.01-1build1) ...
Selecting previously unselected package fonts-wqy-zenhei.
Preparing to unpack .../6-fonts-wqy-zenhei_0.9.45-8_all.deb ...
Unpacking fonts-wqy-zenhei (0.9.45-8) ...
Selecting previously unselected package xfonts-cyrillic.
Preparing to unpack .../7-xfonts-cyrillic_1%3a1.0.5+nmu1_all.deb ...
Unpacking xfonts-cyrillic (1:1.0.5+nmu1) ...
Selecting previously unselected package xfonts-scalable.
Preparing to unpack .../8-xfonts-scalable_1%3a1.0.3-1.3_all.deb ...
Unpacking xfonts-scalable (1:1.0.3-1.3) ...
Setting up fonts-noto-color-emoji (2.047-0ubuntu0.24.04.1) ...
Setting up fonts-wqy-zenhei (0.9.45-8) ...
Setting up fonts-freefont-ttf (20211204+svn4273-2) ...
Setting up fonts-tlwg-loma-otf (1:0.7.3-1) ...
Setting up fonts-ipafont-gothic (00303-21ubuntu1) ...
update-alternatives: using /usr/share/fonts/opentype/ipafont-gothic/ipag.ttf to provide /usr/share/fonts/truetype/fonts-japanese-gothic.ttf (fonts-japanese-gothic.ttf)in auto mode
Setting up xfonts-cyrillic (1:1.0.5+nmu1) ...
Setting up fonts-liberation (1:2.1.5-3) ...
Setting up xfonts-scalable (1:1.0.3-1.3) ...
Setting up fonts-unifont (1:15.1.01-1build1) ...
Processing triggers for fontconfig (2.15.0-1.1ubuntu2) ...
Scanning processes...
Scanning linux images...

Running kernel seems to be up-to-date.

No services need to be restarted.

No containers need to be restarted.

No user sessions are running outdated binaries.

No VM guests are running outdated hypervisor (qemu) binaries on this host.
Downloading Chrome for Testing 147.0.7727.15 (playwright chromium v1217) from https://cdn.playwright.dev/builds/cft/147.0.7727.15/linux64/chrome-linux64.zip
170.4 MiB [====================] 100% 0.0s
Chrome for Testing 147.0.7727.15 (playwright chromium v1217) downloaded to /root/.cache/ms-playwright/chromium-1217
Downloading FFmpeg (playwright ffmpeg v1011) from https://cdn.playwright.dev/dbazure/download/playwright/builds/ffmpeg/1011/ffmpeg-linux.zip
2.3 MiB [====================] 100% 0.0s
FFmpeg (playwright ffmpeg v1011) downloaded to /root/.cache/ms-playwright/ffmpeg-1011
Downloading Chrome Headless Shell 147.0.7727.15 (playwright chromium-headless-shell v1217) from https://cdn.playwright.dev/builds/cft/147.0.7727.15/linux64/chrome-headless-shell-linux64.zip
112 MiB [====================] 100% 0.0s
Chrome Headless Shell 147.0.7727.15 (playwright chromium-headless-shell v1217) downloaded to /root/.cache/ms-playwright/chromium_headless_shell-1217
Note: Workbench disabled — Workbench tests will not be collected.
Note: Package Manager disabled — Package Manager tests will not be collected.
INTERNALERROR> Traceback (most recent call last):
INTERNALERROR>   File "/root/.venv/lib/python3.12/site-packages/_pytest/main.py", line 314, in wrap_session
INTERNALERROR>     config._do_configure()
INTERNALERROR>   File "/root/.venv/lib/python3.12/site-packages/_pytest/config/__init__.py", line 1165, in _do_configure
INTERNALERROR>     self.hook.pytest_configure.call_historic(kwargs=dict(config=self))
INTERNALERROR>   File "/root/.venv/lib/python3.12/site-packages/pluggy/_hooks.py", line 534, in call_historic
INTERNALERROR>     res = self._hookexec(self.name, self._hookimpls.copy(), kwargs, False)
INTERNALERROR>           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
INTERNALERROR>   File "/root/.venv/lib/python3.12/site-packages/pluggy/_manager.py", line 120, in _hookexec
INTERNALERROR>     return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
INTERNALERROR>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
INTERNALERROR>   File "/root/.venv/lib/python3.12/site-packages/pluggy/_callers.py", line 167, in _multicall
INTERNALERROR>     raise exception
INTERNALERROR>   File "/root/.venv/lib/python3.12/site-packages/pluggy/_callers.py", line 121, in _multicall
INTERNALERROR>     res = hook_impl.function(*args)
INTERNALERROR>           ^^^^^^^^^^^^^^^^^^^^^^^^^
INTERNALERROR>   File "/root/.venv/lib/python3.12/site-packages/vip/plugin.py", line 194, in pytest_configure
INTERNALERROR>     session = start_interactive_auth(
INTERNALERROR>               ^^^^^^^^^^^^^^^^^^^^^^^
INTERNALERROR>   File "/root/.venv/lib/python3.12/site-packages/vip/auth.py", line 277, in start_interactive_auth
INTERNALERROR>     browser = _launch_chromium(pw, headless=False)
INTERNALERROR>               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
INTERNALERROR>   File "/root/.venv/lib/python3.12/site-packages/vip/auth.py", line 69, in _launch_chromium
INTERNALERROR>     return pw.chromium.launch(headless=headless)
INTERNALERROR>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
INTERNALERROR>   File "/root/.venv/lib/python3.12/site-packages/playwright/sync_api/_generated.py", line 15118, in launch
INTERNALERROR>     self._sync(
INTERNALERROR>   File "/root/.venv/lib/python3.12/site-packages/playwright/_impl/_sync_base.py", line 115, in _sync
INTERNALERROR>     return task.result()
INTERNALERROR>            ^^^^^^^^^^^^^
INTERNALERROR>   File "/root/.venv/lib/python3.12/site-packages/playwright/_impl/_browser_type.py", line 98, in launch
INTERNALERROR>     await self._channel.send(
INTERNALERROR>   File "/root/.venv/lib/python3.12/site-packages/playwright/_impl/_connection.py", line 69, in send
INTERNALERROR>     return await self._connection.wrap_api_call(
INTERNALERROR>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
INTERNALERROR>   File "/root/.venv/lib/python3.12/site-packages/playwright/_impl/_connection.py", line 559, in wrap_api_call
INTERNALERROR>     raise rewrite_error(error, f"{parsed_st['apiName']}: {error}") from None
INTERNALERROR> playwright._impl._errors.TargetClosedError: BrowserType.launch: Target page, context or browser has been closed
INTERNALERROR> Browser logs:
INTERNALERROR>
INTERNALERROR> ╔════════════════════════════════════════════════════════════════════════════════════════════════╗
INTERNALERROR> ║ Looks like you launched a headed browser without having a XServer running.                     ║
INTERNALERROR> ║ Set either 'headless: true' or use 'xvfb-run <your-playwright-app>' before running Playwright. ║
INTERNALERROR> ║                                                                                                ║
INTERNALERROR> ║ <3 Playwright Team                                                                             ║
INTERNALERROR> ╚════════════════════════════════════════════════════════════════════════════════════════════════╝
INTERNALERROR> Call log:
INTERNALERROR>   - <launching> /root/.cache/ms-playwright/chromium-1217/chrome-linux64/chrome --disable-field-trial-config --disable-background-networking --disable-background-timer-throttling --disable-backgrounding-occluded-windows --disable-back-forward-cache --disable-breakpad --disable-client-side-phishing-detection --disable-component-extensions-with-background-pages --disable-component-update --no-default-browser-check --disable-default-apps --disable-dev-shm-usage --disable-extensions --disable-features=AvoidUnnecessaryBeforeUnloadCheckSync,BoundaryEventDispatchTracksNodeRemoval,DestroyProfileOnBrowserClose,DialMediaRouteProvider,GlobalMediaControls,HttpsUpgrades,LensOverlay,MediaRouter,PaintHolding,ThirdPartyStoragePartitioning,Translate,AutoDeElevate,RenderDocument,OptimizationHints --enable-features=CDPScreenshotNewSurface --allow-pre-commit-input --disable-hang-monitor --disable-ipc-flooding-protection --disable-popup-blocking --disable-prompt-on-repost --disable-renderer-backgrounding --force-color-profile=srgb --metrics-recording-only --no-first-run --password-store=basic --use-mock-keychain --no-service-autorun --export-tagged-pdf --disable-search-engine-choice-screen --unsafely-disable-devtools-self-xss-warnings --edge-skip-compat-layer-relaunch --enable-automation --disable-infobars --disable-search-engine-choice-screen --disable-sync --enable-unsafe-swiftshader --no-sandbox --user-data-dir=/tmp/playwright_chromiumdev_profile-Dg3lC1 --remote-debugging-pipe --no-startup-window
INTERNALERROR>   - <launched> pid=26831
INTERNALERROR>   - [pid=26831][err] [26831:26831:0430/204506.023654:ERROR:ui/ozone/platform/x11/ozone_platform_x11.cc:256] Missing X server or $DISPLAY
INTERNALERROR>   - [pid=26831][err] [26831:26831:0430/204506.023687:ERROR:ui/aura/env.cc:246] The platform failed to initialize.  Exiting.
INTERNALERROR>   - [pid=26831] <gracefully close start>
INTERNALERROR>   - [pid=26831] <kill>
INTERNALERROR>   - [pid=26831] <will force kill>
INTERNALERROR>   - [pid=26831] <process did exit: exitCode=1, signal=null>
INTERNALERROR>   - [pid=26831] starting temporary directories cleanup
INTERNALERROR>   - [pid=26831] finished temporary directories cleanup
INTERNALERROR>   - [pid=26831] <gracefully close end>
(root) root@instance-20260430-1777562952-pwb-01:~# export VIP_TEST_USERNAME=testuser
(root) root@instance-20260430-1777562952-pwb-01:~# export VIP_TEST_PASSWORD=test123
(root) root@instance-20260430-1777562952-pwb-01:~# vip verify --workbench-url https://localhost --headless-auth
Note: Connect disabled — Connect tests will not be collected.
Note: Package Manager disabled — Package Manager tests will not be collected.

>>> Headless auth: authenticating to https://localhost ...
ERROR: Headless auth failed during login: Page.goto: net::ERR_CERT_AUTHORITY_INVALID at https://localhost/
Call log:
  - navigating to "https://localhost/", waiting until "load"
. Rerun with --verbose for details.

(root) root@instance-20260430-1777562952-pwb-01:~# vip verify --workbench-url https://localhost --headless-auth --insecure
Note: Connect disabled — Connect tests will not be collected.
Note: Package Manager disabled — Package Manager tests will not be collected.

>>> Headless auth: authenticating to https://localhost ...
>>> Authentication complete.
========================================================================= test session starts ==========================================================================
collected 100 items / 80 deselected / 1 skipped / 20 selected

.venv/lib/python3.12/site-packages/vip_tests/prerequisites/test_components.py::test_product_server_is_reachable[Workbench] FAILED
.venv/lib/python3.12/site-packages/vip_tests/prerequisites/test_versions.py::test_workbench_version SKIPPED (No Workbench version configured in vip.toml — s...)
.venv/lib/python3.12/site-packages/vip_tests/connect/test_data_sources.py::test_connectivity PASSED
.venv/lib/python3.12/site-packages/vip_tests/cross_product/test_monitoring.py::test_monitoring_configured SKIPPED (Monitoring is not enabled in vip.toml)
.venv/lib/python3.12/site-packages/vip_tests/cross_product/test_resources.py::test_product_health FAILED
.venv/lib/python3.12/site-packages/vip_tests/cross_product/test_ssl.py::test_ssl_certificate_is_valid_for_product[Workbench] FAILED
.venv/lib/python3.12/site-packages/vip_tests/cross_product/test_ssl.py::test_http_redirects_to_https_for_product[Workbench] PASSED
.venv/lib/python3.12/site-packages/vip_tests/cross_product/test_ssl.py::test_tls_12_or_higher_is_enforced_for_product[Workbench] FAILED
.venv/lib/python3.12/site-packages/vip_tests/security/test_auth_policy.py::test_auth_provider PASSED
.venv/lib/python3.12/site-packages/vip_tests/security/test_https.py::test_product_enforces_https[Workbench] PASSED
.venv/lib/python3.12/site-packages/vip_tests/security/test_https.py::test_product_does_not_expose_sensitive_headers[Workbench] PASSED
.venv/lib/python3.12/site-packages/vip_tests/workbench/test_auth.py::test_workbench_login[chromium] SKIPPED (test_auth is not compatible with --headless-auth)
.venv/lib/python3.12/site-packages/vip_tests/workbench/test_data_sources.py::test_data_sources_reachable[chromium] SKIPPED (No data sources configured in vi...)
.venv/lib/python3.12/site-packages/vip_tests/workbench/test_ide_launch.py::test_launch_rstudio[chromium] PASSED
.venv/lib/python3.12/site-packages/vip_tests/workbench/test_ide_launch.py::test_launch_vscode[chromium] PASSED
.venv/lib/python3.12/site-packages/vip_tests/workbench/test_ide_launch.py::test_launch_jupyter[chromium] PASSED
.venv/lib/python3.12/site-packages/vip_tests/workbench/test_ide_launch.py::test_launch_positron[chromium] PASSED
.venv/lib/python3.12/site-packages/vip_tests/workbench/test_packages.py::test_r_repo_configured[chromium] SKIPPED (Package Manager URL is not configured in ...)
.venv/lib/python3.12/site-packages/vip_tests/workbench/test_session_capacity.py::test_launch_sessions_with_the_configured_resource_profile[chromium] PASSED
.venv/lib/python3.12/site-packages/vip_tests/workbench/test_sessions.py::test_session_suspend_resume[chromium]
  ... still running (30s)
  ... still running (60s)
  ... still running (90s)

.venv/lib/python3.12/site-packages/vip_tests/workbench/test_sessions.py::test_session_suspend_resume[chromium] SKIPPED (Session did not return to Active sta...)

=============================================================================== FAILURES ===============================================================================
_____________________________________________________________ test_product_server_is_reachable[Workbench] ______________________________________________________________
test_product_server_is_reachable[Workbench]: an unexpected error occurred: httpcore.ConnectError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable toget local issuer certificate (_ssl.c:1000)
_________________________________________________________________________ test_product_health __________________________________________________________________________
test_product_health: Product health checks failed: Workbench (https://localhost/health-check): [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1000) assert not [{'error': '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1000)', 'ok': False, 'product': 'Workbench', 'status': None, ...}]
_________________________________________________________ test_ssl_certificate_is_valid_for_product[Workbench] _________________________________________________________
test_ssl_certificate_is_valid_for_product[Workbench]: SSL certificate error: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1000) assert '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1000)' is None
_______________________________________________________ test_tls_12_or_higher_is_enforced_for_product[Workbench] _______________________________________________________
test_tls_12_or_higher_is_enforced_for_product[Workbench]: TLS 1.2 handshake reached the server, but the test runner could not verify the server's certificate. This is a certificate-trust issue on the runner, not a TLS-enforcement issue on the server. If the server uses a valid public certificate (e.g. behind an AWS ALB with an ACM cert), set SSL_CERT_FILE to a CA bundle that includes public roots: `/etc/ssl/certs/ca-certificates.crt` on Debian/Ubuntu, `/etc/pki/tls/certs/ca-bundle.crt` on RHEL, or the path produced by `python -m certifi`. Detail: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1000)
================================================== 4 failed, 10 passed, 7 skipped, 80 deselected in 164.54s (0:02:44) ==================================================
(root) root@instance-20260430-1777562952-pwb-01:~# ll /etc/rstudio/rserver.conf
-rw-r--r-- 1 root root 1228 Apr 30 20:39 /etc/rstudio/rserver.conf
(root) root@instance-20260430-1777562952-pwb-01:~# cat $1
^C
(root) root@instance-20260430-1777562952-pwb-01:~# cat /etc/rstudio/rserver.conf
# Posit Workbench Configuration
# Managed by Ansible template: workbench/rstudio-server.conf.j2
# Port and address configuration for Workbench
# Managed by Ansible update: roles/workbench/tasks/ssl_update.yml (SSL)
www-port=443
www-address=0.0.0.0
www-same-site=none
# www-forward-proxy=1

server-health-check-enabled=1
# Trust the forwarded headers
# server-allow-forwarded-headers=1

# Data directory for Workbench
server-data-dir=/var/lib/rstudio-workbench
# Include additional configuration files
# include-config=/etc/rstudio/jupyter.conf

# Job Launcher configuration for Workbench sessions
launcher-sessions-enabled=1
launcher-address=127.0.0.1
launcher-port=5559
launcher-default-cluster=Local
# Managed by Ansible update: roles/workbench/tasks/ssl_update.yml (SSL)
launcher-sessions-callback-address=https://posit.cofer.me

# Admin UI enablement
admin-enabled=1
admin-group=rstudio-admins
# Managed by Ansible update: roles/workbench/tasks/ssl_update.yml (SSL)
ssl-enabled=1
# Managed by Ansible update: roles/workbench/tasks/ssl_update.yml (SSL)
ssl-certificate=/etc/rstudio/posit.cofer.me.crt
# Managed by Ansible update: roles/workbench/tasks/ssl_update.yml (SSL)
ssl-certificate-key=/etc/rstudio/posit.cofer.me.key
(root) root@instance-20260430-1777562952-pwb-01:~# cat /etc/pip/pip.conf
# Date Created: 2025-10-03

[global]
# Use local Package Manager PyPI if available, otherwise Posit's public PyPI
index-url = https://packagemanager.posit.co/pypi/latest/simple/trusted-host = packagemanager.posit.cotimeout = 10
retries = 5
(root) root@instance-20260430-1777562952-pwb-01:~# cat /etc/uv/uv.to^C
(root) root@instance-20260430-1777562952-pwb-01:~# curl -v https://posit.cofer.me/health-check
* Could not resolve host: posit.cofer.me
* Closing connection
curl: (6) Could not resolve host: posit.cofer.me
(root) root@instance-20260430-1777562952-pwb-01:~# curl -v https://localhost/health-check
* Host localhost:443 was resolved.
* IPv6: ::1
* IPv4: 127.0.0.1
*   Trying [::1]:443...
* connect to ::1 port 443 from ::1 port 59718 failed: Connection refused
*   Trying 127.0.0.1:443...
* Connected to localhost (127.0.0.1) port 443
* ALPN: curl offers h2,http/1.1
* TLSv1.3 (OUT), TLS handshake, Client hello (1):
*  CAfile: /etc/ssl/certs/ca-certificates.crt
*  CApath: /etc/ssl/certs
* TLSv1.3 (IN), TLS handshake, Server hello (2):
* TLSv1.2 (IN), TLS handshake, Certificate (11):
* TLSv1.2 (OUT), TLS alert, unknown CA (560):
* SSL certificate problem: unable to get local issuer certificate
* Closing connection
curl: (60) SSL certificate problem: unable to get local issuer certificate
More details here: https://curl.se/docs/sslcerts.html

curl failed to verify the legitimacy of the server and therefore could not
establish a secure connection to it. To learn more about this situation and
how to fix it, please visit the web page mentioned above.
(root) root@instance-20260430-1777562952-pwb-01:~# curl -vk https://localhost/health-check
* Host localhost:443 was resolved.
* IPv6: ::1
* IPv4: 127.0.0.1
*   Trying [::1]:443...
* connect to ::1 port 443 from ::1 port 59726 failed: Connection refused
*   Trying 127.0.0.1:443...
* Connected to localhost (127.0.0.1) port 443
* ALPN: curl offers h2,http/1.1
* TLSv1.3 (OUT), TLS handshake, Client hello (1):
* TLSv1.3 (IN), TLS handshake, Server hello (2):
* TLSv1.2 (IN), TLS handshake, Certificate (11):
* TLSv1.2 (IN), TLS handshake, Server key exchange (12):
* TLSv1.2 (IN), TLS handshake, Server finished (14):
* TLSv1.2 (OUT), TLS handshake, Client key exchange (16):
* TLSv1.2 (OUT), TLS change cipher, Change cipher spec (1):
* TLSv1.2 (OUT), TLS handshake, Finished (20):
* TLSv1.2 (IN), TLS handshake, Finished (20):
* SSL connection using TLSv1.2 / ECDHE-RSA-AES256-GCM-SHA384 / X25519 / RSASSA-PSS
* ALPN: server accepted http/1.1
* Server certificate:
*  subject: C=US; ST=California; L=San Francisco; O=Example Inc.; CN=posit.cofer.me
*  start date: Mar 19 19:08:00 2026 GMT
*  expire date: Mar 19 19:08:00 2027 GMT
*  issuer: CN=My Local CA
*  SSL certificate verify result: unable to get local issuer certificate (20), continuing anyway.
*   Certificate level 0: Public key type RSA (2048/112 Bits/secBits), signed using sha256WithRSAEncryption
* using HTTP/1.x
> GET /health-check HTTP/1.1
> Host: localhost
> User-Agent: curl/8.5.0
> Accept: */*
>
< HTTP/1.1 200 OK
< Server: nginx
< Date: Thu, 30 Apr 2026 20:54:49 GMT
< Content-Type: text/plain
< Content-Length: 185
< Connection: keep-alive
< Expires: Fri, 01 Jan 1990 00:00:00 GMT
< Pragma: no-cache
< Cache-Control: no-cache, no-store, max-age=0, must-revalidate
< X-Content-Type-Options: nosniff
< Strict-Transport-Security: max-age=86400
<
active-sessions: 2
idle-seconds: 0
cpu-percent: 0.0
memory-percent: 8.5
swap-percent: 0.0
load-average: 0.0
license-status: Activated
license-days-left: 6
* Connection #0 to host localhost left intact
license-allow-product-usage: 1(root) root@instance-20260430-1777562952-pwb-01:~# vip verify --connect-url https://localhost --headless-auth --insecure
Note: Workbench disabled — Workbench tests will not be collected.
Note: Package Manager disabled — Package Manager tests will not be collected.
>>> Reusing cached auth session from /root/.vip-auth-cache.json
/root/.venv/lib/python3.12/site-packages/vip/plugin.py:247: UserWarning: VIP: --headless-auth could not mint an API key. API-based tests will likely fail. Try again orset VIP_CONNECT_API_KEY to fix.
  warnings.warn(
========================================================================= test session starts ==========================================================================
collected 100 items / 65 deselected / 1 skipped / 35 selected

.venv/lib/python3.12/site-packages/vip_tests/prerequisites/test_components.py::test_product_server_is_reachable[Connect] FAILED
.venv/lib/python3.12/site-packages/vip_tests/prerequisites/test_versions.py::test_connect_version SKIPPED (No Connect version configured in vip.toml — set c...)
.venv/lib/python3.12/site-packages/vip_tests/connect/test_auth.py::test_connect_login_ui FAILED
.venv/lib/python3.12/site-packages/vip_tests/connect/test_auth.py::test_connect_login_api FAILED
.venv/lib/python3.12/site-packages/vip_tests/connect/test_content_deploy.py::test_deploy_quarto FAILED
.venv/lib/python3.12/site-packages/vip_tests/connect/test_content_deploy.py::test_deploy_plumber ^C

=============================================================================== FAILURES ===============================================================================
______________________________________________________________ test_product_server_is_reachable[Connect] _______________________________________________________________
test_product_server_is_reachable[Connect]: an unexpected error occurred: httpcore.ConnectError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1000)
________________________________________________________________________ test_connect_login_ui _________________________________________________________________________
test_connect_login_ui: an unexpected error occurred: httpcore.ConnectError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1000)
________________________________________________________________________ test_connect_login_api ________________________________________________________________________
test_connect_login_api: an unexpected error occurred: httpcore.ConnectError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1000)
__________________________________________________________________________ test_deploy_quarto __________________________________________________________________________
test_deploy_quarto: an unexpected error occurred: httpcore.ConnectError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1000)
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! KeyboardInterrupt !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
/root/.venv/lib/python3.12/site-packages/httpcore/_backends/base.py:56: KeyboardInterrupt
(to show a full traceback on KeyboardInterrupt use --full-trace)
============================================================= 4 failed, 2 skipped, 65 deselected in 6.72s ==============================================================
Traceback (most recent call last):
  File "/root/.venv/bin/vip", line 10, in <module>
    sys.exit(main())
             ^^^^^^
  File "/root/.venv/lib/python3.12/site-packages/vip/cli.py", line 1147, in main
    args.func(args)
  File "/root/.venv/lib/python3.12/site-packages/vip/cli.py", line 480, in run_verify
    _run_verify_local(args)
  File "/root/.venv/lib/python3.12/site-packages/vip/cli.py", line 574, in _run_verify_local
    result = subprocess.run(cmd, timeout=args.test_timeout)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/subprocess.py", line 550, in run
    stdout, stderr = process.communicate(input, timeout=timeout)
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/subprocess.py", line 1209, in communicate
    stdout, stderr = self._communicate(input, endtime, timeout)
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/subprocess.py", line 2141, in _communicate
    self.wait(timeout=self._remaining_time(endtime))
  File "/usr/lib/python3.12/subprocess.py", line 1264, in wait
    return self._wait(timeout=timeout)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/subprocess.py", line 2047, in _wait
    time.sleep(delay)
KeyboardInterrupt

(root) root@instance-20260430-1777562952-pwb-01:~# vip verify --connect-url https://10.11.234.56 --headless-auth --insecure
Note: Workbench disabled — Workbench tests will not be collected.
Note: Package Manager disabled — Package Manager tests will not be collected.
>>> Reusing cached auth session from /root/.vip-auth-cache.json
/root/.venv/lib/python3.12/site-packages/vip/plugin.py:247: UserWarning: VIP: --headless-auth could not mint an API key. API-based tests will likely fail. Try again orset VIP_CONNECT_API_KEY to fix.
  warnings.warn(
========================================================================= test session starts ==========================================================================
collected 100 items / 65 deselected / 1 skipped / 35 selected

.venv/lib/python3.12/site-packages/vip_tests/prerequisites/test_components.py::test_product_server_is_reachable[Connect] FAILED
.venv/lib/python3.12/site-packages/vip_tests/prerequisites/test_versions.py::test_connect_version SKIPPED (No Connect version configured in vip.toml — set c...)
.venv/lib/python3.12/site-packages/vip_tests/connect/test_auth.py::test_connect_login_ui FAILED
.venv/lib/python3.12/site-packages/vip_tests/connect/test_auth.py::test_connect_login_api FAILED
.venv/lib/python3.12/site-packages/vip_tests/connect/test_content_deploy.py::test_deploy_quarto ^C

=============================================================================== FAILURES ===============================================================================
______________________________________________________________ test_product_server_is_reachable[Connect] _______________________________________________________________
test_product_server_is_reachable[Connect]: an unexpected error occurred: httpcore.ConnectError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1000)
________________________________________________________________________ test_connect_login_ui _________________________________________________________________________
test_connect_login_ui: an unexpected error occurred: httpcore.ConnectError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1000)
________________________________________________________________________ test_connect_login_api ________________________________________________________________________
test_connect_login_api: an unexpected error occurred: httpcore.ConnectError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1000)
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! KeyboardInterrupt !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
/root/.venv/lib/python3.12/site-packages/httpcore/_backends/base.py:56: KeyboardInterrupt
(to show a full traceback on KeyboardInterrupt use --full-trace)
============================================================= 3 failed, 2 skipped, 65 deselected in 4.49s ==============================================================
Traceback (most recent call last):
  File "/root/.venv/bin/vip", line 10, in <module>
    sys.exit(main())
             ^^^^^^
  File "/root/.venv/lib/python3.12/site-packages/vip/cli.py", line 1147, in main
    args.func(args)
  File "/root/.venv/lib/python3.12/site-packages/vip/cli.py", line 480, in run_verify
    _run_verify_local(args)
  File "/root/.venv/lib/python3.12/site-packages/vip/cli.py", line 574, in _run_verify_local
    result = subprocess.run(cmd, timeout=args.test_timeout)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/subprocess.py", line 550, in run
    stdout, stderr = process.communicate(input, timeout=timeout)
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/subprocess.py", line 1209, in communicate
    stdout, stderr = self._communicate(input, endtime, timeout)
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/subprocess.py", line 2141, in _communicate
    self.wait(timeout=self._remaining_time(endtime))
  File "/usr/lib/python3.12/subprocess.py", line 1264, in wait
    return self._wait(timeout=timeout)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/subprocess.py", line 2047, in _wait
    time.sleep(delay)
KeyboardInterrupt

(root) root@instance-20260430-1777562952-pwb-01:~# vip verify --connect-url https://10.11.234.56 --headless-auth --insecure^C
(root) root@instance-20260430-1777562952-pwb-01:~#
(root) root@instance-20260430-1777562952-pwb-01:~#
(root) root@instance-20260430-1777562952-pwb-01:~#
(root) root@instance-20260430-1777562952-pwb-01:~#
(root) root@instance-20260430-1777562952-pwb-01:~#
(root) root@instance-20260430-1777562952-pwb-01:~#
(root) root@instance-20260430-1777562952-pwb-01:~# vip verify --connect-url https://10.11.234.56 --headless-auth --insecure^C
(root) root@instance-20260430-1777562952-pwb-01:~# vip verify --connect-url https://10.11.234.56 --headless-auth --insecure
Note: Workbench disabled — Workbench tests will not be collected.
Note: Package Manager disabled — Package Manager tests will not be collected.
>>> Reusing cached auth session from /root/.vip-auth-cache.json
/root/.venv/lib/python3.12/site-packages/vip/plugin.py:247: UserWarning: VIP: --headless-auth could not mint an API key. API-based tests will likely fail. Try again orset VIP_CONNECT_API_KEY to fix.
  warnings.warn(
========================================================================= test session starts ==========================================================================
collected 100 items / 65 deselected / 1 skipped / 35 selected

.venv/lib/python3.12/site-packages/vip_tests/prerequisites/test_components.py::test_product_server_is_reachable[Connect] FAILED
.venv/lib/python3.12/site-packages/vip_tests/prerequisites/test_versions.py::test_connect_version SKIPPED (No Connect version configured in vip.toml — set c...)
.venv/lib/python3.12/site-packages/vip_tests/connect/test_auth.py::test_connect_login_ui FAILED
.venv/lib/python3.12/site-packages/vip_tests/connect/test_auth.py::test_connect_login_api FAILED
.venv/lib/python3.12/site-packages/vip_tests/connect/test_content_deploy.py::test_deploy_quarto FAILED
.venv/lib/python3.12/site-packages/vip_tests/connect/test_content_deploy.py::test_deploy_plumber FAILED
.venv/lib/python3.12/site-packages/vip_tests/connect/test_content_deploy.py::test_deploy_shiny FAILED
.venv/lib/python3.12/site-packages/vip_tests/connect/test_content_deploy.py::test_deploy_dash FAILED
.venv/lib/python3.12/site-packages/vip_tests/connect/test_content_deploy.py::test_deploy_rmarkdown ^C

=============================================================================== FAILURES ===============================================================================
______________________________________________________________ test_product_server_is_reachable[Connect] _______________________________________________________________
test_product_server_is_reachable[Connect]: an unexpected error occurred: httpcore.ConnectError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1000)
________________________________________________________________________ test_connect_login_ui _________________________________________________________________________
test_connect_login_ui: an unexpected error occurred: httpcore.ConnectError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1000)
________________________________________________________________________ test_connect_login_api ________________________________________________________________________
test_connect_login_api: an unexpected error occurred: httpcore.ConnectError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1000)
__________________________________________________________________________ test_deploy_quarto __________________________________________________________________________
test_deploy_quarto: an unexpected error occurred: httpcore.ConnectError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1000)
_________________________________________________________________________ test_deploy_plumber __________________________________________________________________________
test_deploy_plumber: an unexpected error occurred: httpcore.ConnectError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1000)
__________________________________________________________________________ test_deploy_shiny ___________________________________________________________________________
test_deploy_shiny: an unexpected error occurred: httpcore.ConnectError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1000)
___________________________________________________________________________ test_deploy_dash ___________________________________________________________________________
test_deploy_dash: an unexpected error occurred: httpcore.ConnectError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1000)
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! KeyboardInterrupt !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
/root/.venv/lib/python3.12/site-packages/httpcore/_backends/base.py:56: KeyboardInterrupt
(to show a full traceback on KeyboardInterrupt use --full-trace)
============================================================= 7 failed, 2 skipped, 65 deselected in 10.85s =============================================================
Traceback (most recent call last):
  File "/root/.venv/bin/vip", line 10, in <module>
    sys.exit(main())
             ^^^^^^
  File "/root/.venv/lib/python3.12/site-packages/vip/cli.py", line 1147, in main
    args.func(args)
  File "/root/.venv/lib/python3.12/site-packages/vip/cli.py", line 480, in run_verify
    _run_verify_local(args)
  File "/root/.venv/lib/python3.12/site-packages/vip/cli.py", line 574, in _run_verify_local
    result = subprocess.run(cmd, timeout=args.test_timeout)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/subprocess.py", line 550, in run
    stdout, stderr = process.communicate(input, timeout=timeout)
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/subprocess.py", line 1209, in communicate
    stdout, stderr = self._communicate(input, endtime, timeout)
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/subprocess.py", line 2141, in _communicate
    self.wait(timeout=self._remaining_time(endtime))
  File "/usr/lib/python3.12/subprocess.py", line 1264, in wait
    return self._wait(timeout=timeout)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/subprocess.py", line 2047, in _wait
    time.sleep(delay)
KeyboardInterrupt

(root) root@instance-20260430-1777562952-pwb-01:~# vip verify --connect-url https://^X^C-headless-auth --insecure
(root) root@instance-20260430-1777562952-pwb-01:~# vip verify --package-manager-url https://10.11.230.145 --headless-auth --insecure
Note: Connect disabled — Connect tests will not be collected.
Note: Workbench disabled — Workbench tests will not be collected.
/root/.venv/lib/python3.12/site-packages/vip/plugin.py:219: UserWarning: VIP: --headless-auth was requested but no auth-requiring products (Connect, Workbench) are configured; skipping browser authentication.
  warnings.warn(
========================================================================= test session starts ==========================================================================
collected 100 items / 83 deselected / 1 skipped / 17 selected

.venv/lib/python3.12/site-packages/vip_tests/prerequisites/test_components.py::test_product_server_is_reachable[Package Manager] FAILED
.venv/lib/python3.12/site-packages/vip_tests/prerequisites/test_versions.py::test_package_manager_version SKIPPED (No Package Manager version configured in ...)
.venv/lib/python3.12/site-packages/vip_tests/connect/test_data_sources.py::test_connectivity PASSED
.venv/lib/python3.12/site-packages/vip_tests/cross_product/test_monitoring.py::test_monitoring_configured SKIPPED (Monitoring is not enabled in vip.toml)
.venv/lib/python3.12/site-packages/vip_tests/cross_product/test_resources.py::test_product_health FAILED
.venv/lib/python3.12/site-packages/vip_tests/cross_product/test_ssl.py::test_ssl_certificate_is_valid_for_product[Package Manager] FAILED
.venv/lib/python3.12/site-packages/vip_tests/cross_product/test_ssl.py::test_http_redirects_to_https_for_product[Package Manager] PASSED
.venv/lib/python3.12/site-packages/vip_tests/cross_product/test_ssl.py::test_tls_12_or_higher_is_enforced_for_product[Package Manager] FAILED
.venv/lib/python3.12/site-packages/vip_tests/package_manager/test_private_repos.py::test_private_repos FAILED
.venv/lib/python3.12/site-packages/vip_tests/package_manager/test_repos.py::test_cran_mirror FAILED
.venv/lib/python3.12/site-packages/vip_tests/package_manager/test_repos.py::test_pypi_mirror FAILED
.venv/lib/python3.12/site-packages/vip_tests/package_manager/test_repos.py::test_bioconductor_mirror FAILED
.venv/lib/python3.12/site-packages/vip_tests/package_manager/test_repos.py::test_openvsx_mirror FAILED
.venv/lib/python3.12/site-packages/vip_tests/package_manager/test_repos.py::test_repo_exists FAILED
.venv/lib/python3.12/site-packages/vip_tests/security/test_auth_policy.py::test_auth_provider PASSED
.venv/lib/python3.12/site-packages/vip_tests/security/test_https.py::test_product_enforces_https[Package Manager] PASSED
.venv/lib/python3.12/site-packages/vip_tests/security/test_https.py::test_product_does_not_expose_sensitive_headers[Package Manager] FAILED

=============================================================================== FAILURES ===============================================================================
__________________________________________________________ test_product_server_is_reachable[Package Manager] ___________________________________________________________
test_product_server_is_reachable[Package Manager]: an unexpected error occurred: httpcore.ConnectError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1000)
_________________________________________________________________________ test_product_health __________________________________________________________________________
test_product_health: Product health checks failed: Package Manager (https://10.11.230.145/__api__/status): [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1000) assert not [{'error': '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1000)', 'ok': False, 'product': 'Package Manager', 'status': None, ...}]
______________________________________________________ test_ssl_certificate_is_valid_for_product[Package Manager] ______________________________________________________
test_ssl_certificate_is_valid_for_product[Package Manager]: SSL certificate error: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1000) assert '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1000)' is None
____________________________________________________ test_tls_12_or_higher_is_enforced_for_product[Package Manager] ____________________________________________________
test_tls_12_or_higher_is_enforced_for_product[Package Manager]: TLS 1.2 handshake reached the server, but the test runner could not verify the server's certificate. This is a certificate-trust issue on the runner, not a TLS-enforcement issue on the server. If the server uses a valid public certificate (e.g. behind an AWS ALB with an ACM cert), set SSL_CERT_FILE to a CA bundle that includes public roots: `/etc/ssl/certs/ca-certificates.crt` on Debian/Ubuntu, `/etc/pki/tls/certs/ca-bundle.crt` on RHEL, or the path produced by `python -m certifi`. Detail: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1000)
__________________________________________________________________________ test_private_repos __________________________________________________________________________
test_private_repos: an unexpected error occurred: httpcore.ConnectError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1000)
___________________________________________________________________________ test_cran_mirror ___________________________________________________________________________
test_cran_mirror: an unexpected error occurred: httpcore.ConnectError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1000)
___________________________________________________________________________ test_pypi_mirror ___________________________________________________________________________
test_pypi_mirror: an unexpected error occurred: httpcore.ConnectError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1000)
_______________________________________________________________________ test_bioconductor_mirror _______________________________________________________________________
test_bioconductor_mirror: an unexpected error occurred: httpcore.ConnectError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1000)
_________________________________________________________________________ test_openvsx_mirror __________________________________________________________________________
test_openvsx_mirror: an unexpected error occurred: httpcore.ConnectError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1000)
___________________________________________________________________________ test_repo_exists ___________________________________________________________________________
test_repo_exists: an unexpected error occurred: httpcore.ConnectError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1000)
___________________________________________________ test_product_does_not_expose_sensitive_headers[Package Manager] ____________________________________________________
test_product_does_not_expose_sensitive_headers[Package Manager]: an unexpected error occurred: Failed: Header 'server: Posit Package Manager v2026.04.1' exposes version info. Configure the server to suppress version details.
======================================================= 11 failed, 4 passed, 3 skipped, 83 deselected in 10.17s ========================================================
(root) root@instance-20260430-1777562952-pwb-01:~# vip verify --package-manager-url https^C/10.11.230.145 --headless-auth --insecure
(root) root@instance-20260430-1777562952-pwb-01:~#
```

--- a/src/vip_tests/workbench/test_sessions.py
+++ b/src/vip_tests/workbench/test_sessions.py
@@ -6,6 +6,7 @@ Patterns adapted from test_ide_launch.py.
 
 from __future__ import annotations
 
+import time
 from pathlib import Path
 
 import pytest
@@ -157,27 +158,70 @@ def user_resumes_session(page: Page, session_context: dict):
 
 
 @then("the session reaches Active state again")
-def session_becomes_active_again(page: Page, session_context: dict):
+def session_becomes_active_again(page: Page, workbench_url: str, session_context: dict):
     """Verify the session transitions back to Active state."""
     session_name = session_context["name"]
 
-    # Reload so the DOM reflects the current server state. Unlike the
-    # Active → Suspended check at line ~128, which observes the transition
-    # on the same homepage instance, the resume path requires a reload
-    # because go_back() returns a cached homepage where the "Suspended"
-    # badge has not been re-rendered. After the reload, Workbench's
-    # client-side state polling surfaces the Suspended → Active transition.
-    page.reload(timeout=TIMEOUT_PAGE_LOAD)
+    # Explicitly navigate back to /home. user_resumes_session attempts the
+    # same bounce, but Playwright's goto() races with the IDE iframe loading
+    # on /s/<id> and sometimes leaves the page on the session URL — where the
+    # sidebar shows a stale Suspended badge regardless of the actual server
+    # state. Doing the navigation here, in the observation step, guarantees
+    # we are looking at the real homepage.
+    home_url = workbench_url.rstrip("/") + "/home"
+    page.goto(home_url, timeout=TIMEOUT_PAGE_LOAD)
     expect(page.locator(Homepage.POSIT_LOGO)).to_be_visible(timeout=TIMEOUT_PAGE_LOAD)
 
+    # The Workbench homepage does not auto-poll session state, so a single
+    # locator wait cannot observe the Suspended → Active transition. Reload
+    # periodically inside the overall budget until the Active badge appears.
     session_active = page.locator(Homepage.session_row_status(session_name, "Active"))
+    inner_timeout_ms = 5000
+    deadline = time.time() + (TIMEOUT_SESSION_START / 1000)
+    exc: AssertionError | None = None
+    while True:
+        try:
+            expect(session_active).to_be_visible(timeout=inner_timeout_ms)
+            return  # Active observed
+        except AssertionError as e:
+            exc = e
+            if time.time() >= deadline:
+                break
+            page.reload(timeout=TIMEOUT_PAGE_LOAD)
+            expect(page.locator(Homepage.POSIT_LOGO)).to_be_visible(timeout=TIMEOUT_PAGE_LOAD)
+
+    # Diagnostics before skip: capture what the homepage actually shows so
+    # we can tell whether the session is stuck on Suspended/Starting,
+    # whether the row is missing, or whether the selector format changed.
+    # Temporary — remove once the root cause of #238 is fully understood.
+    diag: list[str] = []
     try:
-        expect(session_active).to_be_visible(timeout=TIMEOUT_SESSION_START)
-    except AssertionError as exc:
-        pytest.skip(
-            f"Session did not return to Active state after resume — "
-            f"suspend/resume may not be supported in this Workbench configuration ({exc})"
-        )
+        diag.append(f"page_url={page.url!r}")
+    except Exception as e:
+        diag.append(f"page_url_error={e!r}")
+    try:
+        row = page.locator(Homepage.session_row(session_name))
+        row_count = row.count()
+        diag.append(f"session_row_count={row_count}")
+        if row_count > 0:
+            badges = row.locator("[aria-label]").all()
+            labels = [b.get_attribute("aria-label") for b in badges[:20]]
+            diag.append(f"row_aria_labels={labels}")
+            diag.append(f"row_text={row.inner_text()[:500]!r}")
+    except Exception as e:
+        diag.append(f"row_inspect_error={e!r}")
+    try:
+        screenshot_dir = Path("report") / "diagnostics"
+        screenshot_dir.mkdir(parents=True, exist_ok=True)
+        screenshot_path = screenshot_dir / f"session_resume_skip_{int(time.time())}.png"
+        page.screenshot(path=str(screenshot_path), full_page=True)
+        diag.append(f"screenshot={screenshot_path}")
+    except Exception as e:
+        diag.append(f"screenshot_error={e!r}")
+    pytest.skip(
+        "Session did not return to Active state after resume — "
+        f"diagnostics: {' | '.join(diag)} | original: {exc}"
+    )
 
 
 @then("the session is cleaned up")

--- a/src/vip_tests/workbench/test_sessions.py
+++ b/src/vip_tests/workbench/test_sessions.py
@@ -161,6 +161,11 @@ def session_becomes_active_again(page: Page, session_context: dict):
     """Verify the session transitions back to Active state."""
     session_name = session_context["name"]
 
+    # Reload so the DOM reflects the current server state — go_back() can leave
+    # a cached page where the "Suspended" badge has not been re-rendered yet.
+    page.reload()
+    expect(page.locator(Homepage.POSIT_LOGO)).to_be_visible(timeout=TIMEOUT_PAGE_LOAD)
+
     session_active = page.locator(Homepage.session_row_status(session_name, "Active"))
     try:
         expect(session_active).to_be_visible(timeout=TIMEOUT_SESSION_START)

--- a/src/vip_tests/workbench/test_sessions.py
+++ b/src/vip_tests/workbench/test_sessions.py
@@ -152,7 +152,16 @@ def user_resumes_session(page: Page, session_context: dict):
         # way to resume.  Click the session name text as a fallback.
         session_row.locator(f"text='{session_name}'").click()
 
-    # Navigate back to homepage to observe the Active state
+    # Wait for the navigation into the session URL to commit before going
+    # anywhere else. Clicking a suspended session navigates to /s/<id> which
+    # triggers the backend resume — but if we navigate away before that
+    # commits, Workbench aborts the start and the session stays Suspended.
+    page.wait_for_url("**/s/**", timeout=TIMEOUT_PAGE_LOAD)
+    page.wait_for_load_state("load", timeout=TIMEOUT_PAGE_LOAD)
+
+    # Navigate back to homepage to observe the Active state. NB: Workbench's
+    # /home may auto-redirect back to the recently-used session — that is OK
+    # for the next step, which observes the session badge in the sidebar.
     page.goto(page.url.split("/s/")[0] + "/home") if "/s/" in page.url else page.go_back()
     expect(page.locator(Homepage.POSIT_LOGO)).to_be_visible(timeout=TIMEOUT_PAGE_LOAD)
 

--- a/src/vip_tests/workbench/test_sessions.py
+++ b/src/vip_tests/workbench/test_sessions.py
@@ -140,15 +140,15 @@ def user_resumes_session(page: Page, session_context: dict):
 
     # Modern Workbench does not expose a one-click launch link on the row
     # for suspended sessions. Active sessions have `a[title='join <name>']`,
-    # but suspended ones only have the session-name link, which opens a
-    # session-details modal that contains a "Launch" button — that button
-    # is what triggers the backend resume. The previous implementation's
-    # `session_row.locator("a").first` was picking up a "Details" link
-    # instead, navigating to /s/<id>/workspaces/ (a management view) and
-    # never resuming the session.
-    name_link = page.locator(Homepage.session_link(session_name)).first
-    expect(name_link).to_be_visible(timeout=TIMEOUT_DIALOG)
-    name_link.click()
+    # but on suspended rows the name is rendered as plain text — clicking
+    # it opens a session-details modal that contains a "Launch" button.
+    # That Launch button is what triggers the backend resume. The previous
+    # implementation's `session_row.locator("a").first` was picking up a
+    # "Details" link instead, navigating to /s/<id>/workspaces/ (a
+    # management view) and never resuming the session.
+    name_text = session_row.locator(Homepage.session_text(session_name))
+    expect(name_text).to_be_visible(timeout=TIMEOUT_DIALOG)
+    name_text.click()
 
     modal = page.locator(Homepage.SESSION_DETAILS_DIALOG)
     expect(modal).to_be_visible(timeout=TIMEOUT_DIALOG)

--- a/src/vip_tests/workbench/test_sessions.py
+++ b/src/vip_tests/workbench/test_sessions.py
@@ -163,7 +163,7 @@ def session_becomes_active_again(page: Page, session_context: dict):
 
     # Reload so the DOM reflects the current server state — go_back() can leave
     # a cached page where the "Suspended" badge has not been re-rendered yet.
-    page.reload()
+    page.reload(timeout=TIMEOUT_PAGE_LOAD)
     expect(page.locator(Homepage.POSIT_LOGO)).to_be_visible(timeout=TIMEOUT_PAGE_LOAD)
 
     session_active = page.locator(Homepage.session_row_status(session_name, "Active"))

--- a/src/vip_tests/workbench/test_sessions.py
+++ b/src/vip_tests/workbench/test_sessions.py
@@ -162,7 +162,11 @@ def session_becomes_active_again(page: Page, session_context: dict):
     session_name = session_context["name"]
 
     # Reload so the DOM reflects the current server state — go_back() can leave
-    # a cached page where the "Suspended" badge has not been re-rendered yet.
+    # a cached page where the "Suspended" badge has not been re-rendered.
+    # After the reload, the Workbench homepage's client-side state polling
+    # surfaces the Suspended → Active transition (same pattern used at line
+    # ~128 above, where Playwright's locator polling observes the
+    # Active → Suspended transition without a reload).
     page.reload(timeout=TIMEOUT_PAGE_LOAD)
     expect(page.locator(Homepage.POSIT_LOGO)).to_be_visible(timeout=TIMEOUT_PAGE_LOAD)
 

--- a/src/vip_tests/workbench/test_sessions.py
+++ b/src/vip_tests/workbench/test_sessions.py
@@ -150,9 +150,13 @@ def user_resumes_session(page: Page, session_context: dict):
     expect(name_text).to_be_visible(timeout=TIMEOUT_DIALOG)
     name_text.click()
 
-    modal = page.locator(Homepage.SESSION_DETAILS_DIALOG)
-    expect(modal).to_be_visible(timeout=TIMEOUT_DIALOG)
-    launch_btn = modal.locator("button:text-is('Launch')")
+    # Wait for the Launch button directly rather than the modal container.
+    # The modal's CSS class varies between Workbench UI versions (the page
+    # object's SESSION_DETAILS_DIALOG selector uses class*='modal-dialog'
+    # which does not match the newer data-slot based dialog markup), but
+    # the button text is stable. A visible Launch button proves the modal
+    # opened.
+    launch_btn = page.locator("button:text-is('Launch')")
     expect(launch_btn).to_be_visible(timeout=TIMEOUT_DIALOG)
     launch_btn.click()
 

--- a/src/vip_tests/workbench/test_sessions.py
+++ b/src/vip_tests/workbench/test_sessions.py
@@ -161,12 +161,12 @@ def session_becomes_active_again(page: Page, session_context: dict):
     """Verify the session transitions back to Active state."""
     session_name = session_context["name"]
 
-    # Reload so the DOM reflects the current server state — go_back() can leave
-    # a cached page where the "Suspended" badge has not been re-rendered.
-    # After the reload, the Workbench homepage's client-side state polling
-    # surfaces the Suspended → Active transition (same pattern used at line
-    # ~128 above, where Playwright's locator polling observes the
-    # Active → Suspended transition without a reload).
+    # Reload so the DOM reflects the current server state. Unlike the
+    # Active → Suspended check at line ~128, which observes the transition
+    # on the same homepage instance, the resume path requires a reload
+    # because go_back() returns a cached homepage where the "Suspended"
+    # badge has not been re-rendered. After the reload, Workbench's
+    # client-side state polling surfaces the Suspended → Active transition.
     page.reload(timeout=TIMEOUT_PAGE_LOAD)
     expect(page.locator(Homepage.POSIT_LOGO)).to_be_visible(timeout=TIMEOUT_PAGE_LOAD)
 

--- a/src/vip_tests/workbench/test_sessions.py
+++ b/src/vip_tests/workbench/test_sessions.py
@@ -206,37 +206,9 @@ def session_becomes_active_again(page: Page, workbench_url: str, session_context
             page.reload(timeout=TIMEOUT_PAGE_LOAD)
             expect(page.locator(Homepage.POSIT_LOGO)).to_be_visible(timeout=TIMEOUT_PAGE_LOAD)
 
-    # Diagnostics before skip: capture what the homepage actually shows so
-    # we can tell whether the session is stuck on Suspended/Starting,
-    # whether the row is missing, or whether the selector format changed.
-    # Temporary — remove once the root cause of #238 is fully understood.
-    diag: list[str] = []
-    try:
-        diag.append(f"page_url={page.url!r}")
-    except Exception as e:
-        diag.append(f"page_url_error={e!r}")
-    try:
-        row = page.locator(Homepage.session_row(session_name))
-        row_count = row.count()
-        diag.append(f"session_row_count={row_count}")
-        if row_count > 0:
-            badges = row.locator("[aria-label]").all()
-            labels = [b.get_attribute("aria-label") for b in badges[:20]]
-            diag.append(f"row_aria_labels={labels}")
-            diag.append(f"row_text={row.inner_text()[:500]!r}")
-    except Exception as e:
-        diag.append(f"row_inspect_error={e!r}")
-    try:
-        screenshot_dir = Path("report") / "diagnostics"
-        screenshot_dir.mkdir(parents=True, exist_ok=True)
-        screenshot_path = screenshot_dir / f"session_resume_skip_{int(time.time())}.png"
-        page.screenshot(path=str(screenshot_path), full_page=True)
-        diag.append(f"screenshot={screenshot_path}")
-    except Exception as e:
-        diag.append(f"screenshot_error={e!r}")
     pytest.skip(
-        "Session did not return to Active state after resume — "
-        f"diagnostics: {' | '.join(diag)} | original: {exc}"
+        f"Session did not return to Active state after resume — "
+        f"suspend/resume may not be supported in this Workbench configuration ({exc})"
     )
 
 

--- a/src/vip_tests/workbench/test_sessions.py
+++ b/src/vip_tests/workbench/test_sessions.py
@@ -132,30 +132,33 @@ def session_becomes_suspended(page: Page, session_context: dict):
 
 @when("the user resumes the session")
 def user_resumes_session(page: Page, session_context: dict):
-    """Click the suspended session link to resume it."""
+    """Open the session details modal and click Launch to resume the session."""
     session_name = session_context["name"]
 
     session_row = page.locator(Homepage.session_row(session_name))
     expect(session_row).to_be_visible(timeout=TIMEOUT_DIALOG)
 
-    # Suspended sessions may not have a clickable link.  Select the session
-    # checkbox and click the session name text to trigger a resume, or fall
-    # back to navigating into the session via any available link.
-    session_link = session_row.locator("a").first
-    if session_link.count() > 0:
-        session_link.click()
-    else:
-        # No link available — select and use the session row action
-        checkbox = page.locator(Homepage.session_checkbox(session_name))
-        checkbox.click()
-        # After selecting a suspended session, the homepage should offer a
-        # way to resume.  Click the session name text as a fallback.
-        session_row.locator(f"text='{session_name}'").click()
+    # Modern Workbench does not expose a one-click launch link on the row
+    # for suspended sessions. Active sessions have `a[title='join <name>']`,
+    # but suspended ones only have the session-name link, which opens a
+    # session-details modal that contains a "Launch" button — that button
+    # is what triggers the backend resume. The previous implementation's
+    # `session_row.locator("a").first` was picking up a "Details" link
+    # instead, navigating to /s/<id>/workspaces/ (a management view) and
+    # never resuming the session.
+    name_link = page.locator(Homepage.session_link(session_name)).first
+    expect(name_link).to_be_visible(timeout=TIMEOUT_DIALOG)
+    name_link.click()
+
+    modal = page.locator(Homepage.SESSION_DETAILS_DIALOG)
+    expect(modal).to_be_visible(timeout=TIMEOUT_DIALOG)
+    launch_btn = modal.locator("button:text-is('Launch')")
+    expect(launch_btn).to_be_visible(timeout=TIMEOUT_DIALOG)
+    launch_btn.click()
 
     # Wait for the navigation into the session URL to commit before going
-    # anywhere else. Clicking a suspended session navigates to /s/<id> which
-    # triggers the backend resume — but if we navigate away before that
-    # commits, Workbench aborts the start and the session stays Suspended.
+    # anywhere else. Navigating away from /s/<id> too quickly causes
+    # Workbench to abort the resume.
     page.wait_for_url("**/s/**", timeout=TIMEOUT_PAGE_LOAD)
     page.wait_for_load_state("load", timeout=TIMEOUT_PAGE_LOAD)
 

--- a/validation_docs/demo-vip-fix-suspend-resume-reload.md
+++ b/validation_docs/demo-vip-fix-suspend-resume-reload.md
@@ -1,9 +1,9 @@
 # Fix: reload page before polling for Active on resume
 
-*2026-05-11T21:20:04Z by Showboat 0.6.1*
-<!-- showboat-id: 0616d658-85a1-4ce6-bb72-404cc36655a4 -->
+*2026-05-12T18:02:59Z by Showboat 0.6.1*
+<!-- showboat-id: fef45b21-a767-4ac9-bc98-936b0adaf83c -->
 
-Issue #238: `test_session_suspend_resume` consistently timed out at 90 seconds waiting for the Active badge to reappear after resume. Root cause: `page.go_back()` in the resume step leaves a stale homepage DOM where the badge still reads "Suspended", so DOM polling never sees the transition to Active. The fix adds `page.reload()` plus a homepage-logo wait at the start of the `session_becomes_active_again` step in `src/vip_tests/workbench/test_sessions.py`. End-to-end verification requires a real Workbench deployment, so this demo proves the change is well-formed: the test still collects, lint passes, and the diff is small and reviewable.
+Issue #238: `test_session_suspend_resume` consistently timed out at 90 seconds waiting for the Active badge to reappear after resume. Root cause: `page.go_back()` in the resume step leaves a stale homepage DOM where the badge still reads "Suspended", so DOM polling never sees the transition to Active. The fix adds `page.reload(timeout=TIMEOUT_PAGE_LOAD)` plus a homepage-logo wait at the start of the `session_becomes_active_again` step in `src/vip_tests/workbench/test_sessions.py`. End-to-end verification requires a real Workbench deployment, so this demo proves the change is well-formed: the test still collects, lint passes, and the diff is small and reviewable.
 
 ```bash
 uv run pytest src/vip_tests/ --collect-only --quiet 2>&1 | tail -1 | sed 's/ in [0-9.]*s//'
@@ -14,13 +14,19 @@ uv run pytest src/vip_tests/ --collect-only --quiet 2>&1 | tail -1 | sed 's/ in 
 ```
 
 ```bash
-uv run ruff check src/vip_tests/workbench/test_sessions.py && uv run ruff format --check src/vip_tests/workbench/test_sessions.py && echo 'ruff: all checks passed'
+uv run ruff check src/vip_tests/workbench/test_sessions.py
 ```
 
 ```output
 All checks passed!
+```
+
+```bash
+uv run ruff format --check src/vip_tests/workbench/test_sessions.py
+```
+
+```output
 1 file already formatted
-ruff: all checks passed
 ```
 
 ```bash
@@ -28,8 +34,8 @@ git diff --stat origin/main -- src/vip_tests/workbench/test_sessions.py
 ```
 
 ```output
- src/vip_tests/workbench/test_sessions.py | 5 +++++
- 1 file changed, 5 insertions(+)
+ src/vip_tests/workbench/test_sessions.py | 9 +++++++++
+ 1 file changed, 9 insertions(+)
 ```
 
 ```bash
@@ -38,16 +44,20 @@ git diff origin/main -- src/vip_tests/workbench/test_sessions.py
 
 ```output
 diff --git a/src/vip_tests/workbench/test_sessions.py b/src/vip_tests/workbench/test_sessions.py
-index d8d067b..3675164 100644
+index d8d067b..5f047d9 100644
 --- a/src/vip_tests/workbench/test_sessions.py
 +++ b/src/vip_tests/workbench/test_sessions.py
-@@ -161,6 +161,11 @@ def session_becomes_active_again(page: Page, session_context: dict):
+@@ -161,6 +161,15 @@ def session_becomes_active_again(page: Page, session_context: dict):
      """Verify the session transitions back to Active state."""
      session_name = session_context["name"]
  
 +    # Reload so the DOM reflects the current server state — go_back() can leave
-+    # a cached page where the "Suspended" badge has not been re-rendered yet.
-+    page.reload()
++    # a cached page where the "Suspended" badge has not been re-rendered.
++    # After the reload, the Workbench homepage's client-side state polling
++    # surfaces the Suspended → Active transition (same pattern used at line
++    # ~128 above, where Playwright's locator polling observes the
++    # Active → Suspended transition without a reload).
++    page.reload(timeout=TIMEOUT_PAGE_LOAD)
 +    expect(page.locator(Homepage.POSIT_LOGO)).to_be_visible(timeout=TIMEOUT_PAGE_LOAD)
 +
      session_active = page.locator(Homepage.session_row_status(session_name, "Active"))

--- a/validation_docs/demo-vip-fix-suspend-resume-reload.md
+++ b/validation_docs/demo-vip-fix-suspend-resume-reload.md
@@ -1,12 +1,12 @@
-# Fix: reload page before polling for Active on resume
+# Fix: resume Workbench session via Launch modal button
 
-*2026-05-12T18:02:59Z by Showboat 0.6.1*
-<!-- showboat-id: fef45b21-a767-4ac9-bc98-936b0adaf83c -->
+*2026-05-13T15:19:34Z by Showboat 0.6.1*
+<!-- showboat-id: 2f460187-9714-4a49-bfbd-86de3751e905 -->
 
-Issue #238: `test_session_suspend_resume` consistently timed out at 90 seconds waiting for the Active badge to reappear after resume. Root cause: `page.go_back()` in the resume step leaves a stale homepage DOM where the badge still reads "Suspended", so DOM polling never sees the transition to Active. The fix adds `page.reload(timeout=TIMEOUT_PAGE_LOAD)` plus a homepage-logo wait at the start of the `session_becomes_active_again` step in `src/vip_tests/workbench/test_sessions.py`. End-to-end verification requires a real Workbench deployment, so this demo proves the change is well-formed: the test still collects, lint passes, and the diff is small and reviewable.
+Issue #238: test_session_suspend_resume was failing because user_resumes_session clicked a 'Details' anchor link on the suspended session row instead of triggering the backend resume. Modern Workbench renders the session name as plain text on suspended rows; clicking it opens a session-details modal containing a 'Launch' button — that button is what initiates the resume. The fix clicks the session name text to open the modal, waits for and clicks the Launch button, then waits for the session URL before returning to /home. In the observation step, session_becomes_active_again now also does an explicit goto /home and retries with page.reload() inside the session-start budget because the Workbench homepage does not auto-poll session state. Validated end-to-end against dev.ganso.lab.staging.posit.team: test_session_suspend_resume[chromium] PASSED.
 
 ```bash
-uv run pytest src/vip_tests/ --collect-only --quiet 2>&1 | tail -1 | sed 's/ in [0-9.]*s//'
+NO_COLOR=1 uv run pytest src/vip_tests/ --collect-only --quiet 2>&1 | tail -1 | sed 's/ in [0-9.]*s//'
 ```
 
 ```output
@@ -34,35 +34,8 @@ git diff --stat origin/main -- src/vip_tests/workbench/test_sessions.py
 ```
 
 ```output
- src/vip_tests/workbench/test_sessions.py | 9 +++++++++
- 1 file changed, 9 insertions(+)
+ src/vip_tests/workbench/test_sessions.py | 89 +++++++++++++++++++++++---------
+ 1 file changed, 65 insertions(+), 24 deletions(-)
 ```
 
-```bash
-git diff origin/main -- src/vip_tests/workbench/test_sessions.py
-```
-
-```output
-diff --git a/src/vip_tests/workbench/test_sessions.py b/src/vip_tests/workbench/test_sessions.py
-index d8d067b..5f047d9 100644
---- a/src/vip_tests/workbench/test_sessions.py
-+++ b/src/vip_tests/workbench/test_sessions.py
-@@ -161,6 +161,15 @@ def session_becomes_active_again(page: Page, session_context: dict):
-     """Verify the session transitions back to Active state."""
-     session_name = session_context["name"]
- 
-+    # Reload so the DOM reflects the current server state — go_back() can leave
-+    # a cached page where the "Suspended" badge has not been re-rendered.
-+    # After the reload, the Workbench homepage's client-side state polling
-+    # surfaces the Suspended → Active transition (same pattern used at line
-+    # ~128 above, where Playwright's locator polling observes the
-+    # Active → Suspended transition without a reload).
-+    page.reload(timeout=TIMEOUT_PAGE_LOAD)
-+    expect(page.locator(Homepage.POSIT_LOGO)).to_be_visible(timeout=TIMEOUT_PAGE_LOAD)
-+
-     session_active = page.locator(Homepage.session_row_status(session_name, "Active"))
-     try:
-         expect(session_active).to_be_visible(timeout=TIMEOUT_SESSION_START)
-```
-
-End-to-end verification — confirming the session badge transitions correctly after resume — requires a real Workbench deployment and is tracked for validation against the staging cluster.
+End-to-end validation against dev.ganso.lab.staging.posit.team on 2026-05-13: test_session_suspend_resume[chromium] PASSED with the session-name text click + Launch modal + reload-retry flow.

--- a/validation_docs/demo-vip-fix-suspend-resume-reload.md
+++ b/validation_docs/demo-vip-fix-suspend-resume-reload.md
@@ -1,0 +1,58 @@
+# Fix: reload page before polling for Active on resume
+
+*2026-05-11T21:20:04Z by Showboat 0.6.1*
+<!-- showboat-id: 0616d658-85a1-4ce6-bb72-404cc36655a4 -->
+
+Issue #238: `test_session_suspend_resume` consistently timed out at 90 seconds waiting for the Active badge to reappear after resume. Root cause: `page.go_back()` in the resume step leaves a stale homepage DOM where the badge still reads "Suspended", so DOM polling never sees the transition to Active. The fix adds `page.reload()` plus a homepage-logo wait at the start of the `session_becomes_active_again` step in `src/vip_tests/workbench/test_sessions.py`. End-to-end verification requires a real Workbench deployment, so this demo proves the change is well-formed: the test still collects, lint passes, and the diff is small and reviewable.
+
+```bash
+uv run pytest src/vip_tests/ --collect-only --quiet 2>&1 | tail -1 | sed 's/ in [0-9.]*s//'
+```
+
+```output
+7/103 tests collected (96 deselected)
+```
+
+```bash
+uv run ruff check src/vip_tests/workbench/test_sessions.py && uv run ruff format --check src/vip_tests/workbench/test_sessions.py && echo 'ruff: all checks passed'
+```
+
+```output
+All checks passed!
+1 file already formatted
+ruff: all checks passed
+```
+
+```bash
+git diff --stat origin/main -- src/vip_tests/workbench/test_sessions.py
+```
+
+```output
+ src/vip_tests/workbench/test_sessions.py | 5 +++++
+ 1 file changed, 5 insertions(+)
+```
+
+```bash
+git diff origin/main -- src/vip_tests/workbench/test_sessions.py
+```
+
+```output
+diff --git a/src/vip_tests/workbench/test_sessions.py b/src/vip_tests/workbench/test_sessions.py
+index d8d067b..3675164 100644
+--- a/src/vip_tests/workbench/test_sessions.py
++++ b/src/vip_tests/workbench/test_sessions.py
+@@ -161,6 +161,11 @@ def session_becomes_active_again(page: Page, session_context: dict):
+     """Verify the session transitions back to Active state."""
+     session_name = session_context["name"]
+ 
++    # Reload so the DOM reflects the current server state — go_back() can leave
++    # a cached page where the "Suspended" badge has not been re-rendered yet.
++    page.reload()
++    expect(page.locator(Homepage.POSIT_LOGO)).to_be_visible(timeout=TIMEOUT_PAGE_LOAD)
++
+     session_active = page.locator(Homepage.session_row_status(session_name, "Active"))
+     try:
+         expect(session_active).to_be_visible(timeout=TIMEOUT_SESSION_START)
+```
+
+End-to-end verification — confirming the session badge transitions correctly after resume — requires a real Workbench deployment and is tracked for validation against the staging cluster.

